tests/test_connections/test_local/test_search_services.py

```diff
@@ -84,6 +86,24 @@ def teardown_class(cls):
         cls.mailbox1.disconnect()
 
 
+class TesAgentSearchAndRegister:
+    """Test that we can register an agent and search him."""
+
+    @classmethod
+    def setup_class(cls):
+        cls.node = LocalNode()
+        cls.public_key_1 = "mailbox1"
+        cls.public_key = "mailbox"
+        cls.connection = OEFLocalConnection(cls.public_key_1, cls.node)
+        cls.mailbox = OEFLocalConnection(cls.public_key, cls.node)
+        cls.mailbox.connect()
+        cls.connection.connect()
+        cls.data_model = DataModel("foobar", attributes=[])
+
+    def test_registering_agent(self):
+        service_description = Description({"foo": 1, "bar": "baz"}, data_model=self.data_model)
+        self.connection.channel
```

Comment:
This is not testing anything

Reply:
Yes ideally, I am registering an agent and then I have to search for this agent in order to complete the test. But we don't have the registering_agent yet. After we merge the register_agent - unregister_agent. I can complete the specific test.